go-controller/pkg/ovn/egressip.go

```diff
@@ -279,24 +293,37 @@ func (oc *Controller) assignEgressIPs(eIP *egressipv1.EgressIP) error {
 			klog.Errorf("Unable to parse provided EgressIP: %s, invalid", egressIP)
 			continue
 		}
-		for i := 0; i < len(eNodes); i++ {
-			klog.V(5).Infof("Attempting assignment on egress node: %+v", eNodes[i])
+		if node := oc.isAnyClusterNodeIP(eIPC); node != nil {
+			eIPRef := kapi.ObjectReference{
+				Kind: "EgressIP",
+				Name: eIP.Name,
+			}
+			oc.recorder.Eventf(
+				&eIPRef,
+				kapi.EventTypeWarning,
+				"UnsupportedRequest",
+				"Egress IP: %v for object EgressIP: %s is the IP address of node: %s, this is unsupported", eIPC, eIP.Name, node.name,
+			)
+			return fmt.Errorf("egress IP: %v is the IP address of node: %s", eIPC, node.name)
+		}
+		for i := 0; i < len(assignableNodes); i++ {
+			klog.V(5).Infof("Attempting assignment on egress node: %+v", assignableNodes[i])
 			if _, exists := existingAllocations[eIPC.String()]; exists {
-				klog.V(5).Infof("EgressIP: %s is already allocated, skipping", eIPC.String())
+				klog.V(5).Infof("EgressIP: %v is already allocated, skipping", eIPC)
 				break
 			}
-			if eNodes[i].tainted {
-				klog.V(5).Infof("Node: %s is already in use by another egress IP for this EgressIP: %s, trying another node", eNodes[i].name, eIP.Name)
+			if assignableNodes[i].tainted {
+				klog.V(5).Infof("Node: %s is already in use by another egress IP for this EgressIP: %s, trying another node", assignableNodes[i].name, eIP.Name)
 				continue
 			}
-			if (utilnet.IsIPv6(eIPC) && eNodes[i].v6Subnet != nil && eNodes[i].v6Subnet.Contains(eIPC)) ||
-				(!utilnet.IsIPv6(eIPC) && eNodes[i].v4Subnet != nil && eNodes[i].v4Subnet.Contains(eIPC)) {
-				eNodes[i].tainted, oc.eIPAllocator[eNodes[i].name].allocations[eIPC.String()] = true, true
+			if (utilnet.IsIPv6(eIPC) && assignableNodes[i].v6Subnet != nil && assignableNodes[i].v6Subnet.Contains(eIPC)) ||
+				(!utilnet.IsIPv6(eIPC) && assignableNodes[i].v4Subnet != nil && assignableNodes[i].v4Subnet.Contains(eIPC)) {
```

Comment:
similarly, could do just

	if (assignableNodes[i].v6Subnet != nil && assignableNodes[i].v6Subnet.Contains(eIPC)) ||
		(assignableNodes[i].v4Subnet != nil && assignableNodes[i].v4Subnet.Contains(eIPC)) {

You're allowed to ask if an IPv6 subnet contains an IPv4 IP or vice versa. (The answer is just guaranteed to be "no".)